sdk/samples/aztables/sample_batch.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/data/aztables"
+)
+
+type MyEntity struct {
+	aztables.Entity
+	Price       float32
+	Inventory   int32
+	ProductName string
+	OnSale      bool
+}
+
+func Sample_Batching() {
+	accountName, ok := os.LookupEnv("TABLES_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("TABLES_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	accountKey, ok := os.LookupEnv("TABLES_PRIMARY_STORAGE_ACCOUNT_KEY")
+	if !ok {
+		panic("TABLES_PRIMARY_STORAGE_ACCOUNT_KEY could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.table.core.windows.net/%s", accountName, "tableName")
+
+	cred, err := aztables.NewSharedKeyCredential(accountName, accountKey)
+	check(err)
+	client, err := aztables.NewClient(serviceURL, cred, nil)
+	check(err)
+
+	batch := make([]aztables.TransactionAction, 4)
+
+	entity1 := MyEntity{
+		Entity: aztables.Entity{
+			PartitionKey: "pk001",
+			RowKey:       "rk001",
+		},
+		Price:       3.99,
+		Inventory:   10,
+		ProductName: "Pens",
+		OnSale:      false,
+	}
+	marshalled, err := json.Marshal(entity1)
+	check(err)
+	batch[0] = aztables.TransactionAction{
+		ActionType: aztables.Add,
+		Entity:     marshalled,
+	}
+
+	entity2 := MyEntity{
+		Entity: aztables.Entity{
+			PartitionKey: "pk001",
+			RowKey:       "rk002",
+		},
+		Price:       19.99,
+		Inventory:   15,
+		ProductName: "Calculators",
+		OnSale:      false,
+	}
+	marshalled, err = json.Marshal(entity2)
+	check(err)
+	batch[1] = aztables.TransactionAction{
+		ActionType: aztables.UpdateMerge,
+		Entity:     marshalled,
+	}
+
+	entity3 := MyEntity{
+		Entity: aztables.Entity{
+			PartitionKey: "pk001",
+			RowKey:       "rk003",
+		},
+		Price:       0.99,
+		Inventory:   150,
+		ProductName: "Pens",
+		OnSale:      true,
+	}
+	marshalled, err = json.Marshal(entity3)
```

Comment:
I'm sure this API has already been through the wringer _but_ is there anything in our library that just automatically handles marshalling the entity? Or are these steps always required?

Reply:
These steps are always required